extensions/postgres/error_checker.go

```diff
@@ -4,20 +4,22 @@ import (
 	"context"
 	"database/sql"
 	"errors"
+	"fmt"
 	"github.com/lib/pq"
 )
 
-// ErrDupEntry indicates a duplicate primary key i.e. the row already exists,
+// ErrDupEntryCode indicates a duplicate primary key i.e. the row already exists,
 // check http://www.postgresql.org/docs/9.3/static/errcodes-appendix.html
-const ErrDupEntry = "23505"
+const ErrDupEntryCode = pq.ErrorCode("23505")
```

Comment:
IIUC, unique_violation will occur when attempting to insert a new row with values ("namespace", "process_id", "2") into the db, while the db already has a row with values ("namespace", "process_id", "1").

Then I think we need to modify the logic here later to support the idReusePolicy?

Reply:
Oh I see the comment.